pkg/storage/hotplug/hotplug_test.go

```diff
@@ -0,0 +1,332 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2025 The KubeVirt Contributors
+ *
+ */
+package hotplug
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/kubevirt/fake"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	libvmistatus "kubevirt.io/kubevirt/pkg/libvmi/status"
+)
+
+type addVolFunc func(diskName, pvcName string, diskOpts ...libvmi.DiskOption) libvmi.Option
+
+var _ = Describe("Volume Hotplug", func() {
+	var virtClient *kubecli.MockKubevirtClient
+	var virtFakeClient *fake.Clientset
+
+	BeforeEach(func() {
+		virtClient = kubecli.NewMockKubevirtClient(gomock.NewController(GinkgoT()))
+		virtFakeClient = fake.NewSimpleClientset()
+
+		// Set up mock client
```

Comment:
Is the fake client not sufficient?

Reply:
`HandleDeclarativeVolumes` takes a `kubecli.KubevirtClient` which is what we typically pass around to functions that need a client. What's the downside?